apps/src/templates/publishDialog/PublishDialog.jsx

```diff
@@ -0,0 +1,54 @@
+import React, { Component, PropTypes } from 'react';
+import Radium from 'radium';
+import Dialog, { Body, styles as dialogStyles } from '../Dialog';
+import PendingButton from '../PendingButton';
+import LegacyButton, { BUTTON_TYPES } from '../LegacyButton';
+import i18n from '@cdo/locale';
+
+class PublishDialog extends Component {
+  static propTypes = {
+    isOpen: PropTypes.bool.isRequired,
+    isPublishPending: PropTypes.bool.isRequired,
+    onClose: PropTypes.func.isRequired,
+    onConfirmPublish: PropTypes.func.isRequired,
+    projectId: PropTypes.string,
+    projectType: PropTypes.string,
+  }
+
+  confirm = () => {
+    this.props.onConfirmPublish(
+      this.props.projectId,
+      this.props.projectType,
+    );
+  };
+
+  close = () => this.props.onClose();
+
+  render() {
+    return (
+      <Dialog
+        title={i18n.publishToPublicGallery()}
+        isOpen={this.props.isOpen}
+        handleClose={this.close}
```

Comment:
My understanding is that `this` needs to be bound here: https://facebook.github.io/react/docs/handling-events.html